linkerd/stack/src/watch.rs

```diff
@@ -0,0 +1,96 @@
+use crate::{NewService, Param, Service};
+use std::task::{Context, Poll};
+use tokio::sync::watch;
+use tracing::Instrument;
+
+pub trait UpdateWatch<T> {
+    type Service: Clone + Default;
```

Comment:
i put the `Clone` and `Default` bounds here because we'll get a much more obvious type error if they're missing

Reply:
we probably should also put the `Send` + `Sync` bounds here since this trait is only used by `NewSpawnWatch` anyway...